src/Microsoft.Health.Fhir.SqlServer/Features/Storage/SqlServerFhirDataStore.cs

```diff
@@ -143,7 +152,15 @@ public async Task<UpsertOutcome> UpsertAsync(ResourceWrapper resource, WeakETag
                     switch (e.Number)
                     {
                         case SqlErrorCodes.PreconditionFailed:
-                            throw new PreconditionFailedException(string.Format(Core.Resources.ResourceVersionConflict, weakETag?.VersionId));
```

Comment:
Is it worth to add new error code for this case?
I'm not sure logic to check etag==null is capturing all cases

Reply:
I was thinking about it... we currently have SQL error codes mapped to HTTP error codes in [SqlErrorCodes.cs](https://github.com/microsoft/healthcare-shared-components/blob/master/src/Microsoft.Health.SqlServer/Features/Storage/SqlErrorCodes.cs) (in the hsc repo).

We would need to change the convention since both scenarios throw a PreconditionFailedException in this case. Thoughts?

Reply:
I haven't touch Upsert code for awhile so, if you think, condition you have covers all the cases and draws clear line, I'm fine with that.
I'm not insisting, I just see something what potentially can be problematic, and ask question : is this enough?

Reply:
My bad, it turns out the spec requires a different error code for each scenario. This check is no longer needed and has been removed in 564cf6f. 

I created a table to summarize the different scenarios and their corresponding result codes [here](https://microsoft-my.sharepoint.com/:x:/g/personal/rotodd_microsoft_com1/EU9rTJGc8hdCjGe7BIFp3CEBNvU0p2QKIXqLfXJlTsD8aQ?e=CtSRmi).